beta.go

```diff
@@ -0,0 +1,58 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package mathext
+
+import (
+	"math"
+)
+
+// Beta returns the value of the complete beta function B(a, b).
```

Comment:
Beta returns the value of the complete beta function B(a, b). It is defined as
 Γ(a)Γ(b) / Γ(a+b)

Special cases are:
 B(a,b) returns +Inf if a xor b is 0
 B(a,b) returns  NaN if a or b is < 0
 B(a,b) returns  NaN if a or b is NaN.